README.MD
# Obras

## URL
[https://obras.buenosaires.gob.ar/](https://obras.buenosaires.gob.ar/)

## Observatorio de obras 

Proyecto de visualización y análisis de la información de las obras del Gobierno de la Ciudad.

El proyecto es exclusivamente frontend. Toma la información de unos CSV estáticos y los representa. 

Se trata de una aplicación enteramente de frontend (todos archivos estáticos), con diferentes vistas. Los datos se toman desde los archivos almacenados en el Portal de Datos a través de la API [csv-to api](http://git-asi.buenosaires.gob.ar/usuarioQA/ssdeciudadinteligentedggedi-15-compromisos-csv-to-api)

No tiene formularios, submits a backend con información ni acceso a base de datos.

Las páginas y los charts que aquí se generan, serán embebidos en un iframe dentro de BuenosAires.gob.ar.

Tomar como referencia ante dudas, la configuración de: [Compromisos de gobierno](https://compromisos-site.buenosaires.gob.ar/) dado que la aplicación presenta iguales características.

Está construído como Single Page Application basado en Angular JS.

## Documentación

* Deploy: ver INSTALL.MD y UPGRADE.MD
* Changelog: ver CHANGELOG.MD 

=======

## Probar el proyecto de manera standalone para desarrollo
* Clonar el proyecto
* Archivo de configuración: En /dist duplicar el archivo config.js.example con el nombre config.js.
* Instalar NodeJS: [Node Js Oficial](http://nodejs.org)
* Instalar http-server
`npm install http-server -g`
* Ir a Directorio /dist y ejecutar servidor
`cd dist`
`http-server ./`
* Abrir navegador 
Dirección: `http://localhost:8080`

## Para desarrolladores
* Archivo de configuración: En /app duplicar el archivo config.js.example con el nombre config.js.
* Correr la aplicación desde /app -> Hacer los cambios en /app y con live reloading se actualizará en http://localhost:10000
`grunt server`
* Compilar
`grunt build`

## Probar los compilados. 
* Levanta la aplicación en http://localhost:10000 desde /dist
`grunt server:dist`

## Instalar en producción
* Hacer clone y/o pull del proyecto 
* Apuntar la configuración del web server a la carpera /dist
* En /dist duplicar el archivo config.js.example con el nombre config.js.
* Navegar hacia la url

This project is generated with [yo angular generator](https://github.com/yeoman/generator-angular)
version 0.15.1.

## Build & development

Run `grunt` for building and `grunt serve` for preview.

## Testing

Running `grunt test` will run the unit tests with karma.

=======

## Notas sobre documentación de USIG

[Documentación de USIG-JS](http://servicios.usig.buenosaires.gob.ar/usig-js/3.1/doc/)

[Tenes demos acá](http://servicios.usig.buenosaires.gob.ar/usig-js/3.1/demos/)

[Y Ejemplos acá:](http://servicios.usig.buenosaires.gob.ar/usig-js/3.1/ejemplos/)

Es bastante importante que incluyan los componentes desde servicios.usig.buenosaires.gob.ar en lugar de copiarlos porque de lo contrario los bug-fixes no van a impactar en forma automática.
Si querés usar Leaflet acá te paso un ejemplo de cómo cargarías la cartografía de USIG:

```javascript
usig.Layers = {
    'amba_con_transporte': {
        url: '//tiles1.usig.buenosaires.gob.ar/mapcache/tms/1.0.0/amba_con_transporte_3857@GoogleMapsCompatible/{z}/{x}/{y}.png',
        format: 'tms',
        builder: 'tms',
        baseLayer: true,
        options: {
            maxZoom: 18,
            minZoom: 9,
            attribution:'USIG (<a href="http://www.buenosaires.gob.ar" target="_blank">GCBA</a>), © <a href="http://www.openstreetmap.org/copyright/en" target="_blank">OpenStreetMap</a> (ODbL)',
            tms: true
        },
    },
}

    var map = L.map('map', {
        center: L.latLng(-34.607497, -58.443832),
        zoom: 11,
        attributionControl: false
    });

    L.tileLayer(usig.Layers.amba_con_transporte.url, usig.Layers.amba_con_transporte.options).addTo(map);
    var lezama = L.marker([-34.628913, -58.369937]).addTo(map);
    lezama.bindPopup("<h3>Palacio Lezama</h3>");
    
    $.ajax ({
        type:'GET',
        dataType:'json',
        url: URL_DEL_SERVICIO_QUE_DEVUELVE_GEOJSON,
        error: function() {
            alert('No se pudieron cargar los datos');
        },
        success: function(data) {
            L.geoJson(data, {
                onEachFeature: popup /* Funcion a definir */,
                pointToLayer: styleMarker /* Estilo (opcional) */
            }).addTo(map);
        }
    });  

```  

=======

Bienvenido al Repositorio de Entregas de los proyectos a la ASI, denominado GIT.
La definición de como se entrega y que documentos deben entregar se encuentra en la carpeta denominada "Instructivos-Documentos ASI" donde encontrarán un instructivo del GIT que explica el proceso y donde subir cada archivo. 
Adicionalmente se encuentran los templates de guía para completar la documentación del proyecto para avanzar con las tareas y actividades de la ASI como Documento de Arquitectura, Manual de instalación, Documento de Alcance y funcionamiento, etc.

Es obligatorio que contemplen que  el sistema operativo que deben utilizar es **RedHat 6.5**. Puede descargar la versión gratuita del mismo (Centos 6.5) desde el siguiente enlace [Descargar](https://github.com/2creatives/vagrant-centos/releases/download/v6.5.1/centos65-x86_64-20131205.box)

Recuerden que **ninguno** de los ambientes productivos cuentan con acceso a internet. Si para el correcto funcionamiento de la aplicación es necesario comunicarse con servidores externos los mismos debe enumerarse en el manual de instalación indicando dominio/puerto que debe estar habilitado justificando su uso en cada caso. 




CHANGELOG.MD
# CHANGELOG

* 2018-03-28 - v3.6
  * Limpieza de código

* 2018-03-28 - v3.5
  * Se removieron archivos y dependencias innecesarias.

* 2018-03-28 - v3.4
  * Actualización de archivos de documentación

* 2018-03-28 - v3.3
  * Actualización de archivos de documentación en `documents` e `INSTALL.MD`.
  * Se quitaron funcionalidades no requeridas.
  * Se actualizó la librería `pym.js` a su versión más estable.

* 2018-03-05 - v3.2
  * Actualización de archivos de documentación en `documents`

* 2018-01-31 - v3.1
  * Actualización de archivos de documentación

* 2018-01-15 - v3.0
  * Nuevos filtros y charts en la home

* 2017-10-23 - v2.3
  * Nuevos estilos para indicadores
  * Nuevo indicador de empleados

* 2017-09-21 - v2.2
  * Especificar el pull en el UPGRADE.md

* 2017-09-20 - v2.1
  * Fixes chart de totales.
  * Validación espacios en blanco nuevo dataset para comunas.

* 2017-09-13 - v2.0
  * Adaptación para visualizar cerca de 1000 obras de distintos Ministerios.
  * Nuevos filtros.
  * Nuevos charts.
  * Mejoras en el tooltip.
  * Nuevo diseño más amplio.

* 2017-08-16 - v1.2
  * Nueva vista de mapas.
  * Mejoras en diseño general.
  * Nuevos campos en el dataset.

* 2016-12-01 - v1.1
  * Nueva vista de buscador avanzado de obras.
  * Nueva vista por entorno.

* 2016-10-30 - v1.0
  * Versión inicial MDUYT.
  * Home: sankey.
  * Vista individual de obra.



Gruntfile.js
// Generated on 2016-10-03 using generator-angular 0.15.1
"use strict";

// # Globbing
// for performance reasons we're only matching one level down:
// 'test/spec/{,*/}*.js'
// use this if you want to recursively match all subfolders:
// 'test/spec/**/*.js'

module.exports = function(grunt) {
  // Time how long tasks take. Can help when optimizing build times
  require("time-grunt")(grunt);

  // Automatically load required Grunt tasks
  require("jit-grunt")(grunt, {
    useminPrepare: "grunt-usemin",
    ngtemplates: "grunt-angular-templates",
    cdnify: "grunt-google-cdn"
  });

  // Configurable paths for the application
  var appConfig = {
    app: require("./bower.json").appPath || "app",
    dist: "dist"
  };

  // Define the configuration for all the tasks
  grunt.initConfig({
    // Project settings
    yeoman: appConfig,

    // Watches files for changes and runs tasks based on the changed files
    watch: {
      bower: {
        files: ["bower.json"],
        tasks: ["wiredep"]
      },
      js: {
        files: ["<%= yeoman.app %>/scripts/{,*/}*.js"],
        tasks: ["newer:jshint:all", "newer:jscs:all"],
        options: {
          livereload: "<%= connect.options.livereload %>"
        }
      },
      jsTest: {
        files: ["test/spec/{,*/}*.js"],
        tasks: ["newer:jshint:test", "newer:jscs:test", "karma"]
      },
      styles: {
        files: ["<%= yeoman.app %>/styles/{,*/}*.css"],
        tasks: ["newer:copy:styles", "postcss"]
      },
      gruntfile: {
        files: ["Gruntfile.js"]
      },
      livereload: {
        options: {
          livereload: "<%= connect.options.livereload %>"
        },
        files: [
          "<%= yeoman.app %>/{,*/}*.html",
          ".tmp/styles/{,*/}*.css",
          "<%= yeoman.app %>/images/{,*/}*.{png,jpg,jpeg,gif,webp,svg}"
        ]
      }
    },

    // The actual grunt server settings
    connect: {
      options: {
        port: 10000,
        // Change this to '0.0.0.0' to access the server from outside.
        hostname: "localhost",
        livereload: 35729
      },
      livereload: {
        options: {
          open: true,
          middleware: function(connect) {
            return [
              connect.static(".tmp"),
              connect().use(
                "/bower_components",
                connect.static("./bower_components")
              ),
              connect().use("/app/styles", connect.static("./app/styles")),
              connect.static(appConfig.app)
            ];
          }
        }
      },
      test: {
        options: {
          port: 9001,
          middleware: function(connect) {
            return [
              connect.static(".tmp"),
              connect.static("test"),
              connect().use(
                "/bower_components",
                connect.static("./bower_components")
              ),
              connect.static(appConfig.app)
            ];
          }
        }
      },
      dist: {
        options: {
          open: true,
          base: "<%= yeoman.dist %>"
        }
      }
    },

    // Make sure there are no obvious mistakes
    jshint: {
      options: {
        jshintrc: ".jshintrc",
        reporter: require("jshint-stylish")
      },
      all: {
        src: ["Gruntfile.js", "<%= yeoman.app %>/scripts/{,*/}*.js"]
      },
      test: {
        options: {
          jshintrc: "test/.jshintrc"
        },
        src: ["test/spec/{,*/}*.js"]
      }
    },

    // Make sure code styles are up to par
    jscs: {
      options: {
        config: ".jscsrc",
        verbose: true
      },
      all: {
        src: ["Gruntfile.js", "<%= yeoman.app %>/scripts/{,*/}*.js"]
      },
      test: {
        src: ["test/spec/{,*/}*.js"]
      }
    },

    // Empties folders to start fresh
    clean: {
      dist: {
        files: [
          {
            dot: true,
            src: [
              ".tmp",
              "<%= yeoman.dist %>/{,*/}*",
              "!<%= yeoman.dist %>/.git{,*/}*"
            ]
          }
        ]
      },
      server: ".tmp"
    },

    // Add vendor prefixed styles
    postcss: {
      options: {
        processors: [
          require("autoprefixer-core")({ browsers: ["last 1 version"] })
        ]
      },
      server: {
        options: {
          map: true
        },
        files: [
          {
            expand: true,
            cwd: ".tmp/styles/",
            src: "{,*/}*.css",
            dest: ".tmp/styles/"
          }
        ]
      },
      dist: {
        files: [
          {
            expand: true,
            cwd: ".tmp/styles/",
            src: "{,*/}*.css",
            dest: ".tmp/styles/"
          }
        ]
      }
    },

    // Automatically inject Bower components into the app
    wiredep: {
      app: {
        src: ["<%= yeoman.app %>/app.html"],
        ignorePath: /\.\.\//
      },
      test: {
        devDependencies: true,
        src: "<%= karma.unit.configFile %>",
        ignorePath: /\.\.\//,
        fileTypes: {
          js: {
            block: /(([\s\t]*)\/{2}\s*?bower:\s*?(\S*))(\n|\r|.)*?(\/{2}\s*endbower)/gi,
            detect: {
              js: /'(.*\.js)'/gi
            },
            replace: {
              js: "'{{filePath}}',"
            }
          }
        }
      }
    },

    // Renames files for browser caching purposes
    filerev: {
      dist: {
        src: [
          "<%= yeoman.dist %>/scripts/{,*/}*.js",
          "<%= yeoman.dist %>/styles/{,*/}*.css",
          "<%= yeoman.dist %>/images/{,*/}*.{png,jpg,jpeg,gif,webp}",
          "!<%= yeoman.dist %>/scripts/pym.v1.min.js",
          "!<%= yeoman.dist %>/images/layers*",
          "!<%= yeoman.dist %>/images/marker*"
        ]
      }
    },

    // Reads HTML for usemin blocks to enable smart builds that automatically
    // concat, minify and revision files. Creates configurations in memory so
    // additional tasks can operate on them
    useminPrepare: {
      html: "<%= yeoman.app %>/app.html",
      options: {
        dest: "<%= yeoman.dist %>",
        flow: {
          html: {
            steps: {
              js: ["concat", "uglifyjs"],
              css: ["cssmin"]
            },
            post: {}
          }
        }
      }
    },

    // Performs rewrites based on filerev and the useminPrepare configuration
    usemin: {
      html: ["<%= yeoman.dist %>/{,*/}*.html"],
      css: ["<%= yeoman.dist %>/styles/{,*/}*.css"],
      js: ["<%= yeoman.dist %>/scripts/{,*/}*.js"],
      options: {
        assetsDirs: [
          "<%= yeoman.dist %>",
          "<%= yeoman.dist %>/images",
          "<%= yeoman.dist %>/styles"
        ],
        patterns: {
          js: [
            [
              /(images\/[^''""]*\.(png|jpg|jpeg|gif|webp|svg))/g,
              "Replacing references to images"
            ]
          ]
        }
      }
    },

    // The following *-min tasks will produce minified files in the dist folder
    // By default, your `index.html`'s <!-- Usemin block --> will take care of
    // minification. These next options are pre-configured if you do not wish
    // to use the Usemin blocks.
    // cssmin: {
    //   dist: {
    //     files: {
    //       '<%= yeoman.dist %>/styles/main.css': [
    //         '.tmp/styles/{,*/}*.css'
    //       ]
    //     }
    //   }
    // },
    // uglify: {
    //   dist: {
    //     files: {
    //       '<%= yeoman.dist %>/scripts/scripts.js': [
    //         '<%= yeoman.dist %>/scripts/scripts.js'
    //       ]
    //     }
    //   }
    // },
    // concat: {
    //   dist: {}
    // },

    imagemin: {
      dist: {
        files: [
          {
            expand: true,
            cwd: "<%= yeoman.app %>/images",
            src: "{,*/}*.{png,jpg,jpeg,gif}",
            dest: "<%= yeoman.dist %>/images"
          }
        ]
      }
    },

    svgmin: {
      dist: {
        files: [
          {
            expand: true,
            cwd: "<%= yeoman.app %>/images",
            src: "{,*/}*.svg",
            dest: "<%= yeoman.dist %>/images"
          }
        ]
      }
    },

    htmlmin: {
      dist: {
        options: {
          collapseWhitespace: true,
          conservativeCollapse: true,
          collapseBooleanAttributes: true,
          removeCommentsFromCDATA: true
        },
        files: [
          {
            expand: true,
            cwd: "<%= yeoman.dist %>",
            src: ["*.html"],
            dest: "<%= yeoman.dist %>"
          }
        ]
      }
    },

    ngtemplates: {
      dist: {
        options: {
          module: "obrasMduytApp",
          htmlmin: "<%= htmlmin.dist.options %>",
          usemin: "scripts/scripts.js"
        },
        cwd: "<%= yeoman.app %>",
        src: "views/{,*/}*.html",
        dest: ".tmp/templateCache.js"
      }
    },

    // ng-annotate tries to make the code safe for minification automatically
    // by using the Angular long form for dependency injection.
    ngAnnotate: {
      dist: {
        files: [
          {
            expand: true,
            cwd: ".tmp/concat/scripts",
            src: "*.js",
            dest: ".tmp/concat/scripts"
          }
        ]
      }
    },

    // Replace Google CDN references
    cdnify: {
      dist: {
        html: ["<%= yeoman.dist %>/*.html"]
      }
    },

    // Copies remaining files to places other tasks can use
    copy: {
      dist: {
        files: [
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>",
            src: [
              "*.{ico,png,txt}",
              "*.html",
              ".htaccess",
              "images/{,*/}/{,*/}/*",
              "styles/fonts/{,*/}*.*"
            ]
          },
          {
            expand: true,
            cwd: ".tmp/images",
            dest: "<%= yeoman.dist %>/images",
            src: ["generated/*"]
          },
          {
            expand: true,
            cwd: "bower_components/bootstrap/dist",
            src: "fonts/*",
            dest: "<%= yeoman.dist %>"
          },
          {
            expand: true,
            cwd: "bower_components/BAstrap/fonts",
            src: "*",
            dest: "<%= yeoman.dist %>/styles/fonts"
          },
          {
            expand: true,
            cwd: "bower_components/leaflet/dist/images",
            src: "*",
            dest: "<%= yeoman.dist %>/images"
          },
          {
            expand: true,
            cwd: "bower_components/flexslider/fonts",
            src: "*",
            dest: "<%= yeoman.dist %>/styles/fonts"
          },
          {
            expand: true,
            cwd: "bower_components/pym.js/dist",
            src: "pym.v1.min.js",
            dest: "<%= yeoman.dist %>/scripts"
          },
          {
            expand: true,
            cwd: "app",
            src: "config.js.example",
            dest: "<%= yeoman.dist %>"
          },
          {
            expand: true,
            cwd: "<%= yeoman.app %>/geo",
            src: "*.geojson",
            dest: "<%= yeoman.dist %>/geo"
          },
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>/fonts",
            src: [".htaccess"]
          },
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>/geo",
            src: [".htaccess"]
          },
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>/images",
            src: [".htaccess"]
          },
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>/scripts",
            src: [".htaccess"]
          },
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>/styles",
            src: [".htaccess"]
          },
          {
            expand: true,
            dot: true,
            cwd: "<%= yeoman.app %>",
            dest: "<%= yeoman.dist %>/styles/fonts",
            src: [".htaccess"]
          }
        ]
      },
      styles: {
        expand: true,
        cwd: "<%= yeoman.app %>/styles",
        dest: ".tmp/styles/",
        src: "{,*/}*.css"
      }
    },

    // Run some tasks in parallel to speed up the build process
    concurrent: {
      server: ["copy:styles"],
      test: ["copy:styles"],
      dist: ["copy:styles", "imagemin", "svgmin"]
    },

    // Test settings
    karma: {
      unit: {
        configFile: "test/karma.conf.js",
        singleRun: true
      }
    }
  });

  grunt.registerTask(
    "serve",
    "Compile then start a connect web server",
    function(target) {
      if (target === "dist") {
        global.runningDist = true;
        return grunt.task.run(["build", "connect:dist:keepalive"]);
      }

      grunt.task.run([
        "clean:server",
        "wiredep",
        "concurrent:server",
        "postcss:server",
        "connect:livereload",
        "watch"
      ]);
    }
  );

  grunt.registerTask(
    "server",
    'DEPRECATED TASK. Use the "serve" task instead',
    function(target) {
      grunt.log.warn(
        "The `server` task has been deprecated. Use `grunt serve` to start a server."
      );
      grunt.task.run(["serve:" + target]);
    }
  );

  grunt.registerTask("test", [
    "clean:server",
    "wiredep",
    "concurrent:test",
    "postcss",
    "connect:test",
    "karma"
  ]);

  grunt.registerTask("build", [
    "clean:dist",
    "wiredep",
    "useminPrepare",
    "concurrent:dist",
    "postcss",
    "ngtemplates",
    "concat",
    "ngAnnotate",
    "copy:dist",
    "cdnify",
    "cssmin",
    "uglify",
    "filerev",
    "usemin",
    "htmlmin"
  ]);

  grunt.registerTask("default", [
    "newer:jshint",
    "newer:jscs",
    "test",
    "build"
  ]);
};
